rfq/manager.go

```diff
@@ -982,11 +983,19 @@ func (m *Manager) AssetMatchesSpecifier(ctx context.Context,
 			return false, err
 		}
 
+		specifierGK := specifier.UnwrapGroupKeyToPtr()
+
+		// If the group lookup failed, let's see if the provided assetID
+		// is the hash of the group key, which is used by the sender to
```

Comment:
This is a bit confusing, why aren't we just passing around the group key isntead of this special case where a hash of it can be provided?

Reply:
If the look up fails, doesn't that mean we haven't verified this asset at all yet? That should have happened during funding, when we send the transition proofs of each of the assets.

Reply:
Hmm, so see my other inline comment. So we do look up the value we put into the dummy ID again here.
Not sure what's better, doing an explicit hash or just use the x-coordinates of the group key directly.

But what we can do in any case is do the `groupHash == id` comparison **before** querying the database. Because if that matches, we **know** there isn't an asset ID in the DB. That would probably also allow us to get rid of the previous commit.